File: SplitSmart/backend/models/expense.js
import mongoose from 'mongoose';

const expenseSchema = new mongoose.Schema({
  groupId: { type: mongoose.Schema.Types.ObjectId, ref: 'Group', required: true },
  title: String,
  amount: Number,
  paidBy: String,
  date: Date,
  notes: String
});

export default mongoose.model('Expense', expenseSchema); 

File: SplitSmart/lib/models/expense.dart
class Expense {
  final String id;
  final String groupId;
  final String title;
  final double amount;
  final String paidBy;
  final DateTime date;
  final String? notes;

  Expense({
    required this.id,
    required this.groupId,
    required this.title,
    required this.amount,
    required this.paidBy,
    required this.date,
    this.notes,
  });
}
